src/net_processing.h

```diff
@@ -94,6 +94,9 @@ class PeerLogicValidation final : public CValidationInterface, public NetEventsI
 private:
     int64_t m_stale_tip_check_time; //!< Next time to check for stale tip
 
+    /** Whether we've completed initial sync yet, for determining when to turn
+      * on extra block-relay peers. */
+    bool m_initial_sync_finished{false};
```

Comment:
Should this ever be set back to false? For example, if we were offline for a week and we know we're catching up.

Reply:
I think this would mean that we stop using block-relay-only peer rotation when our tip is stale, which might be when we want these connections happening the most?  It becomes arguable whether we should just rely on stale-tip-checking + full outbound peer rotation at that point, but I think we would want to carefully reason about our protections in that scenario.